build-system/pr-check.js

```diff
@@ -430,6 +430,32 @@ function runYarnLockfileCheck() {
   }
 }
 
+/**
+ * Returns true if this is a PR build for a greenkeeper branch.
+ */
+function isGreenkeeperPrBuild() {
+  return (process.env.TRAVIS_EVENT_TYPE == 'pull_request') &&
+      (process.env.TRAVIS_PULL_REQUEST_BRANCH.indexOf('greenkeeper/') == 0);
+}
+
+/**
+ * Returns true if this is a push build for a greenkeeper branch.
+ */
+function isGreenkeeperPushBuild() {
+  return (process.env.TRAVIS_EVENT_TYPE == 'push') &&
+      (process.env.TRAVIS_BRANCH.indexOf('greenkeeper/') == 0);
+}
+
+/**
+ * Returns true if this is a push build for a lockfile update on a greenkeeper
+ * branch.
+ */
+function isGreenkeeperLockfilePushBuild() {
+  return isGreenkeeperPushBuild() &&
+      (process.env.TRAVIS_COMMIT_MESSAGE.indexOf(
+          'chore(package): update lockfile') == 0);
```

Comment:
is there a less fragile way to detect this?

Reply:
Not that I could think of. Here is the complete commit message (it spans multiple lines):
```
commit d211e9b136a4ee934bdcf16ee44e5579cc80b41e
Author: greenkeeperio-bot <support@greenkeeper.io>
Date:   Wed Feb 7 18:12:57 2018 +0000

    chore(package): update lockfile
    
    https://npm.im/greenkeeper-lockfile
```
The reason I'm not just searching for `update lockfile` somewhere in the message is that I wanted to eliminate commit messages of the type `revert: chore(package): update lockfile` which are generated by github in a revert PR.

Reply:
I'll keep an eye on the commits during the next revert (whenever it might be!) and see if there's a better way to do this.